### ISSUE TITLE
Conflicting TF coordinates published for cobalt

### ISSUE THREAD

**Issue by hmmwhatsthisdo** [PalouseRobosub/robosub-2018#324]
When running Gazebo, the RViz bridge, and the Localization engine, the TF-published coordinates of `cobalt` (relative to `world`) jump between the Localization engine's output and the origin point.

```sh
roslaunch robosub_simulator gazebo.launch &
roslaunch robosub rviz.launch &
roslaunch robosub localization.launch &
rqt &
```
Inside of `rqt`, open a Plot window and add the following topics to the view:
* `/rviz/cobalt/pose/pose/position`
* `/real/position`
* `/tf/transforms[0]/transform/translation` 

Instead of matching `/rviz/cobalt/pose/pose/position` (the output from the localization engine, in this case), `/tf/transforms[0]/transform/translation` is oscillating between that and zero.

![image](https://user-images.githubusercontent.com/2093321/31869697-923111ea-b75e-11e7-9bbc-9ce321369ced.png)

**Comment by hmmwhatsthisdo** [comment]
Trial-and-error suggests the `sub_tf_broadcaster` node is publishing the incorrect TF data. Looking at [sub_tf_broadcaster.py](https://github.com/PalouseRobosub/robosub/blob/5b12824860db82ec148dc19e888470c2644f6af1/src/sensors/sub_tf_broadcaster.py) and the rviz launchfile seems to suggest it is indeed setting `cobalt`'s location to be 0 relative to `world`.

At this point, I'm unsure if that broadcaster is still needed, or if we want to simply augment its functionality. (@irwineffect may have more knowledge on its original/current purpose) Some ideas that come to mind are:
- Publish updated/zeroed transformation data for `cobalt` if no other transformation data is received for _n_ seconds.
- The above, but with the position of the `cobalt` model according to the `simulator_bridge` (assuming the bridge node is active)

**Comment by irwineffect** [comment]
Yeah, [sub_tf_broadcaster.py](https://github.com/PalouseRobosub/robosub/blob/5b12824860db82ec148dc19e888470c2644f6af1/src/sensors/sub_tf_broadcaster.py) is definitely broadcasting bad data, when I wrote it I just had it report (0,0,0) for the position. At the time, I wasn't aware the localization engine also publishes TF data. I think sub_tf_broadcast.py should take in the position as reported by the localization engine (via it's `/position` topic or whatever) and use that as the tf data for the `cobalt` frame. I think it would be a good idea to have a completely different tf frame for the actual position as reported by the simulator, perhaps `cobalt_sim` frame? Once we have that, we can have rviz display the sub where the localization says it is, and then have a "ghosted" sub displaying the simulator's reported position.

**Comment by skallaher** [comment]
I like the way that @irwineffect suggests to do this. If it doesn't receive a point from localization, we can default to (0, 0, 0) or some other reasonable default.

**Comment by skallaher** [comment]
@hmmwhatsthisdo, is this fixed by #327?

**Comment by hmmwhatsthisdo** [comment]
Yes, this is - I thought merging #327 would have closed this, but apparently I forgot to reference it in the PR so GH didn't close automatically.
